### PR TITLE
JDI working with Appium now

### PR DESCRIPTION
**Pull request by dexguitar**

### CODE REVIEW

**Comment by mf-code** [comment]
[4688] - https://transfer.sh/ScvXi/jdi-light-html-tests.tar.gz

**Comment by mf-code** [comment]
[4688] - https://transfer.sh/DqAgb/jdi-performance.tar.gz

File: jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java [hunk 1]
```diff
@@ -21,40 +27,40 @@
...
 	// @UI("[ui=jdi-text]") public static Text jdiText;
 
-    @UI("[value*='Red Button']") public static Button redButton;
+    @UI("[value*='Red Button']") public static com.epam.jdi.mobile.ui.html.elements.common.Button redButton;
     // @UI("input[type=button].red") public static Button redButton;
-    @UI("[value*='Disabled Button']") public static Button disabledButtonInput;
+    @UI("[value*='Disabled Button']") public static com.epam.jdi.mobile.ui.html.elements.common.Button disabledButtonInput;
     // @UI("input[type=button][disabled]") public static Button disabledButtonInput;
-    public static Button blueButton, dblClickButton, rightClickButton;
+    public static com.epam.jdi.mobile.ui.html.elements.common.Button blueButton, dblClickButton, rightClickButton;
```
Comment (codacy-review) [inline jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 24
- Added 9
           

Complexity increasing per file
==============================
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverTypes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196&bid=16426530)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.epam.jdi.mobile.driver.get;
+
+public class MobileDriver {
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java]: Codacy Issue found: [The utility class name 'MobileDriver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 3]
```diff
@@ -44,92 +53,84 @@
...
-    public static Map<String,String> CAPABILITIES_FOR_FF = new HashMap<>();
-    public static Map<String,String> CAPABILITIES_FOR_EDGE = new HashMap<>();
-    public static Map<String,String> CAPABILITIES_FOR_OPERA = new HashMap<>();
-    public static Map<String,String> COMMON_CAPABILITIES = new HashMap<>();
+    public static Map<String, String> CAPABILITIES_FOR_IE = new HashMap<>();
+    public static Map<String, String> CAPABILITIES_FOR_CHROME = new HashMap<>();
+    public static Map<String, String> CAPABILITIES_FOR_FF = new HashMap<>();
+    public static Map<String, String> CAPABILITIES_FOR_EDGE = new HashMap<>();
+    public static Map<String, String> CAPABILITIES_FOR_OPERA = new HashMap<>();
+    public static Map<String, String> CAPABILITIES_FOR_MOBILE = new HashMap<>();
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 4]
```diff
@@ -168,84 +171,108 @@ public static void setUp(String name, JAction action) {
...
+        setUp("Mobile: deviceName: emulator-5554",
+                () -> cap.setCapability(MobileCapabilityType.DEVICE_NAME, "emulator-5554"));
+        setUp("Mobile: platformName: Android",
+                () -> cap.setCapability(MobileCapabilityType.PLATFORM_NAME, "Android"));
+        setUp("Mobile: browserName: Chrome",
+                () -> cap.setCapability(MobileCapabilityType.BROWSER_NAME, MobileBrowserType.CHROME));
+        CAPABILITIES_FOR_MOBILE.forEach(cap::setCapability);
     };
 
+    private static String getBrowserSizeOption() {
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Avoid unused private methods such as 'getBrowserSizeOption()'.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 5]
```diff
@@ -168,84 +171,108 @@ public static void setUp(String name, JAction action) {
...
-            () -> cap.setCapability(ACCEPT_SSL_CERTS, true));
-        // Capabilities from settings
-        CAPABILITIES_FOR_IE.forEach(cap::setCapability);
-    };
-    public static JAction1<EdgeOptions> EDGE_OPTIONS = cap -> {
-
-    };
-    public static JAction1<OperaOptions> OPERA_OPTIONS = cap -> {
-
+    public static JAction1<DesiredCapabilities> MOBILE_OPTIONS = cap -> {
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/WebDriverFactory.java [hunk 6]
```diff
@@ -5,6 +5,7 @@
 import com.epam.jdi.tools.func.JFunc;
 import com.epam.jdi.tools.map.MapArray;
 import com.epam.jdi.tools.pairs.Pair;
+import io.appium.java_client.AppiumDriver;
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/WebDriverFactory.java]: Codacy Issue found: [Avoid unused imports such as 'io.appium.java_client.AppiumDriver'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverInfos.java [hunk 7]
```diff
@@ -1,72 +1,39 @@
 package com.epam.jdi.mobile.driver.get;
 
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.MutableCapabilities;
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverInfos.java]: Codacy Issue found: [Avoid unused imports such as 'org.openqa.selenium.MutableCapabilities'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 8]
```diff
@@ -44,92 +53,84 @@
...
-                maximizeScreen(driver);
-            else
-                driver.manage().window().maximize();
-        }
-        return driver;
-    };
-
-    private static WebDriver setBrowserSizeForMac(WebDriver driver,
-            int width, int height) {
+    private static void setBrowserSizeForMac(WebDriver driver,
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Avoid unused private methods such as 'setBrowserSizeForMac(WebDriver,int,int)'.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java [hunk 9]
```diff
@@ -21,40 +27,40 @@
...
     @WaitTimeout(4)
-    public static Button ghostButton, suspendButton;
+    public static com.epam.jdi.mobile.ui.html.elements.common.Button ghostButton, suspendButton;
```
Comment (codacy-review) [inline jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by mf-code** [comment]
[4700] - Could not save metadata Could not save metadata Could not save metadata

**Comment by mf-code** [comment]
[4700] - Could not save metadata Could not save metadata Could not save metadata

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 24
- Added 9
           

Complexity increasing per file
==============================
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverTypes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196&bid=16426530)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java (hunk 8 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Avoid unused private methods such as 'setBrowserSizeForMac(WebDriver,int,int)'.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/elements/init/PageFactory.java [hunk 10]
```diff
@@ -30,6 +30,7 @@
 import static com.epam.jdi.mobile.driver.WebDriverFactory.getDriver;
 import static com.epam.jdi.mobile.driver.WebDriverFactory.useDriver;
 import static com.epam.jdi.mobile.driver.get.DriverData.DRIVER_NAME;
+import static com.epam.jdi.mobile.driver.get.DriverInfos.MOBILE_INFO;
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/elements/init/PageFactory.java]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.mobile.driver.get.DriverInfos.MOBILE_INFO'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java]: Codacy Issue found: [The utility class name 'MobileDriver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java (hunk 5 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java (hunk 9 above)
Comment (codacy-review) [inline jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java (hunk 1 above)
Comment (codacy-review) [inline jdi-light-html-tests/src/main/java/io/github/com/pages/HtmlElementsPage.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java (hunk 3 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java (hunk 4 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Avoid unused private methods such as 'getBrowserSizeOption()'.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverInfos.java (hunk 7 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverInfos.java]: Codacy Issue found: [Avoid unused imports such as 'org.openqa.selenium.MutableCapabilities'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by mf-code** [comment]
[4711] - Could not save metadata Could not save metadata Could not save metadata

**Comment by mf-code** [comment]
[4711] - Could not save metadata Could not save metadata Could not save metadata

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 33
- Added 3
           

Complexity increasing per file
==============================
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverTypes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196&bid=16426530)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java]: Codacy Issue found: [The utility class name 'MobileDriver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 12]
```diff
@@ -18,118 +25,112 @@
...
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Avoid unused private methods such as 'setBrowserSizeForMac(WebDriver,int,int)'.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java (hunk 5 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by mf-code** [comment]
[4713] - Could not save metadata Could not save metadata Could not save metadata

**Comment by mf-code** [comment]
[4713] - Could not save metadata Could not save metadata Could not save metadata

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 33
- Added 2
           

Complexity increasing per file
==============================
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverTypes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196&bid=16426530)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java]: Codacy Issue found: [The utility class name 'MobileDriver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 13]
```diff
@@ -168,84 +182,123 @@ public static void setUp(String name, JAction action) {
...
-        setUp("IE: SUPPORTS_JAVASCRIPT",
-            () -> cap.is(SUPPORTS_JAVASCRIPT));
-        setUp("IE: ACCEPT_SSL_CERTS: true",
-            () -> cap.setCapability(ACCEPT_SSL_CERTS, true));
-        // Capabilities from settings
-        CAPABILITIES_FOR_IE.forEach(cap::setCapability);
-    };
-    public static JAction1<EdgeOptions> EDGE_OPTIONS = cap -> {
-
+    public static JAction1<DesiredCapabilities> MOBILE_OPTIONS = cap -> {
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by mf-code** [comment]
[4716] - Could not save metadata Could not save metadata Could not save metadata

**Comment by mf-code** [comment]
[4716] - Could not save metadata Could not save metadata Could not save metadata

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 39
- Added 1
           

Complexity increasing per file
==============================
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverTypes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196&bid=16426530)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java]: Codacy Issue found: [The utility class name 'MobileDriver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by mf-code** [comment]
[4718] - Could not save metadata Could not save metadata Could not save metadata

**Comment by mf-code** [comment]
[4718] - Could not save metadata Could not save metadata Could not save metadata

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 39
- Added 1
           

Complexity increasing per file
==============================
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverTypes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196&bid=16426530)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java]: Codacy Issue found: [The utility class name 'MobileDriver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by mf-code** [comment]
[4728] - https://transfer.sh/12uVeL/jdi-light-html-tests.tar.gz

**Comment by mf-code** [comment]
[4728] - https://transfer.sh/dA7GR/jdi-performance.tar.gz

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 38
- Added 6
           

Complexity increasing per file
==============================
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java  1
- jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverTypes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196&bid=16426530)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 14]
```diff
@@ -157,95 +222,92 @@ public static void setUp(String name, JAction action) {
...
-    };
-    public static JAction1<EdgeOptions> EDGE_OPTIONS = cap -> {
-
-    };
-    public static JAction1<OperaOptions> OPERA_OPTIONS = cap -> {
-
-    };
+    private static WebDriver maximizeScreen(WebDriver driver) {
+        try {
+            switch (getOs()) {
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [A switch statement does not contain a break](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/MobileDriver.java]: Codacy Issue found: [The utility class name 'MobileDriver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 15]
```diff
@@ -157,95 +222,92 @@ public static void setUp(String name, JAction action) {
...
+                    java.awt.Dimension screenSize = getDefaultToolkit().getScreenSize();
+                    setBrowserSizeForMac(driver, (int) screenSize.getWidth(), (int) screenSize.getHeight());
+            }
+            return driver;
+        } catch (Exception ex) {
+            throw new RuntimeException("Failed to maximize window: ", ex);
+        }
+    }
 
+    private static String getBrowserSizeOption() {
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Avoid unused private methods such as 'getBrowserSizeOption()'.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 16]
```diff
@@ -104,32 +171,30 @@ public static OsTypes getOs() {
         if (groups.size() == 2)
             driver.manage().window().setSize(new Dimension(parseInt(groups.get(0)), parseInt(groups.get(1))));
         else {
-            if (getOs().equals(MAC))
+            if (getOs().equals(MAC) && !DRIVER_NAME.equals("appium")) {
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java (hunk 14 above)
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

File: jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java [hunk 17]
```diff
@@ -157,95 +222,92 @@ public static void setUp(String name, JAction action) {
...
+            switch (getOs()) {
+                case WIN:
+                    driver.manage().window().maximize();
+                case MAC:
+                    java.awt.Dimension screenSize = getDefaultToolkit().getScreenSize();
+                    setBrowserSizeForMac(driver, (int) screenSize.getWidth(), (int) screenSize.getHeight());
+            }
+            return driver;
+        } catch (Exception ex) {
+            throw new RuntimeException("Failed to maximize window: ", ex);
```
Comment (codacy-review) [inline jdi-light-mobile/src/main/java/com/epam/jdi/mobile/driver/get/DriverData.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=4973196)

**Comment by mf-code** [comment]
[4730] - https://transfer.sh/9uWd9/jdi-performance.tar.gz

**Comment by mf-code** [comment]
[4730] - https://transfer.sh/4NCTt/jdi-light-html-tests.tar.gz

**Pull request merged**